feat: Add Pipedrive Writer connector

Jkarage opened the pull request:
This Adds Pipedrive Writer Connector.

Test output:
<img width="1204" alt="Screenshot 2024-10-16 at 15 23 06" src="https://github.com/user-attachments/assets/9520b1d6-4253-4dca-a6b9-f35caa225574">

jlimatampersand commented on providers/pipedrive/parse.go:
```diff
@@ -0,0 +1,34 @@
+)
+
+// nextRecordsURL builds the next-page url func.
+func nextRecordsURL(url *urlbuilder.URL) common.NextPageFunc {
```
Can we add some link to documentation where it shows the data structure? Or example data so that we can see where `additional_data` `pagination` `more_items_in_collection` are coming from. This will also guide us when there is need for modification.

jlimatampersand commented on providers/pipedrive/read.go:
```diff
@@ -0,0 +1,54 @@
+	"github.com/amp-labs/connectors/common/urlbuilder"
+)
+
+func (c *Connector) Read(ctx context.Context, config common.ReadParams) (*common.ReadResult, error) {
```
please add a link to API documentation in a comment.

jlimatampersand commented on providers/pipedrive/write.go:
```diff
@@ -0,0 +1,51 @@
+	// Other fields.
+}
+
+func (c *Connector) Write(ctx context.Context, config common.WriteParams) (*common.WriteResult, error) {
```
Same here, please add link to API documentation for write.

jlimatampersand reviewed (changes requested):
Please resolve my comments, mostly looking good. A few things
- Change the title of this PR since this also adds read function. 
- In comment, add link to documentations for each method. 
- Add test result for read. 

Feel free to ping me directly or mention me in the slack channel when done.

Jkarage commented on scripts/oauth/token.go:
```diff
@@ -272,7 +272,7 @@ func setup() *OAuthApp {
 
 	substitutions, err := registry.GetMap(SubstitutionsFieldName)
 	if err != nil {
-		slog.Warn("no substitutions, ensure that the provider info doesn't have any {{variables}}", err)
```
This suddenly caused test failings(building was failing), Updated this to resolve the  it.

Jkarage commented:
@jlimatampersand I have rebased this to the updated Read connector, so this will be a dedicated Write Connector PR. The Read [PR](https://github.com/amp-labs/connectors/pull/1076).  I have added the documentation links as you recommended as well.

jlimatampersand commented on scripts/oauth/token.go:
```diff
@@ -272,7 +272,7 @@ func setup() *OAuthApp {
 	substitutions, err := registry.GetMap(SubstitutionsFieldName)
 	if err != nil {
-		slog.Warn("no substitutions, ensure that the provider info doesn't have any {{variables}}", err)
+		slog.Warn("no substitutions, ensure that the provider info doesn't have any variables", slog.String("error: ", err.Error()))
```
slog.Warn("no substitutions, ensure that the provider info doesn't have any variables", "error", err)

jlimatampersand reviewed (approved):
Looks good. Feel free to merge after slog.Warn("...") is resolved
